pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -472,11 +472,15 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 	// If we need to (re-)create the pod sandbox, everything will need to be
 	// killed and recreated, and init containers should be purged.
 	if createPodSandbox {
-		if !shouldRestartOnFailure(pod) && attempt != 0 {
+		if !shouldRestartOnFailure(pod) && attempt != 0 && len(podStatus.ContainerStatuses) != 0 {
```

Comment:
Should we also add check `len(podStatus.InitContainerStatuses) != 0`?

Reply:
This is a kubelet's internal type; there's no initcontianerstatuses.

Reply:
The unit tests added show that this would work for init containers too.